pypeit/specutils/pypeit_loaders.py

```diff
@@ -195,7 +195,10 @@ def pypeit_spec1d_loader(filename, extract=None, fluxed=True, strict=True, **kwa
         _ext, _cal = sobj.best_ext_match(extract=extract, fluxed=fluxed)
         _wave, _flux, _ivar, _gpm = sobj.get_box_ext(fluxed=_cal) if _ext == 'BOX' \
                                         else sobj.get_opt_ext(fluxed=_cal)
-        _wave, _flux, _ivar = _enforce_monotonic_wavelengths(_wave, _flux, _ivar, strict=strict)
+        if not np.all(_gpm):
+            msgs.warn(f'Ignoring {np.sum(np.logical_not(_gpm))} masked pixels.')
+        _wave, _flux, _ivar = _enforce_monotonic_wavelengths(_wave[_gpm], _flux[_gpm], _ivar[_gpm],
+                                                             strict=strict)
```

Comment:
What happens if there are no good pixels (_i.e._, all are zero for some reason)?  Will `_enforce_monotonic_wavelengths()` as written choke on empty input arrays?  Maybe add an edge-case check to bypass the enforcement in this case?

Reply:
Good catch!  What do you think of my fix:
```
if not np.any(_gpm):
    msgs.warn(f'Spectrum {sobj.NAME} is fully masked and will be ignored!')
    continue
```

Reply:
That looks good.